### PR TITLE
ES-2500: Fix Corda CLI installer and S3 download script task

### PR DESCRIPTION
**Pull request by anton-subbotin**
After removing PF4J dependency from Corda CLI (https://github.com/corda/corda-runtime-os/pull/6169), few issues have been spotted:
  
- task `cliS3Download` got broken and referring to outdated paths
- unintentional changes in the cli install archive contents

See details in Jira: https://r3-cev.atlassian.net/browse/ES-2500

This PR:
- fixes `cliS3Download` task and the way it is supposed to be used during the release publishing process
- Brings back CLI installer behaviour from 5.2: zip archive includes only `corda-cli.jar` and two install scripts. Install script creates runnable script in the user environment as a part of the install process

However, it is arguable that the `cliS3Download` task, and the `templateScript/corda-cli-downloader.sh` artifact script produced by it, still have any user value. See https://github.com/corda/corda-runtime-os/pull/1990 as of Sep 8, 2022

### CODE REVIEW

File: tools/corda-cli/build.gradle [hunk 1]
```diff
@@ -51,19 +50,23 @@ tasks.register('copyCliFiles') {
 tasks.register("cliInstallArchive", Zip) {
     dependsOn tasks.named('copyCliFiles')
     from "$buildDir/cli"
-    include '*'
```
Comment (anton-subbotin) [inline tools/corda-cli/build.gradle:54]: This reverts the unintentional change by https://github.com/corda/corda-runtime-os/pull/6169, which resulted in install archive including also older versions of `corda-cli.sh` and `corda-cli.ps1` scripts from the `scripts/runScripts` directory - these have migrated from the `corda-cli-plugin-host` to support running Corda CLI from the build directory without installing, as is being referred to in few places

File: tools/corda-cli/build.gradle [hunk 2]
```diff
@@ -51,19 +50,23 @@ tasks.register('copyCliFiles') {
...
-    include '*/*'
+    include '*.jar', 'install.*'
```
Comment (anton-subbotin) [inline tools/corda-cli/build.gradle:53]: This reverts the unintentional change by https://github.com/corda/corda-runtime-os/pull/6169, which resulted in install archive including also older versions of `corda-cli.sh` and `corda-cli.ps1` scripts from the `scripts/runScripts` directory - these have migrated from the `corda-cli-plugin-host` to support running Corda CLI from the build directory without installing, as is being referred to in few places.

In previous released versions, corda-cli-installer archive doesn't include run scripts; rather, the installer script is responsible for creation of the run scripts on the fly.

File: tools/corda-cli/scripts/installScripts/install.sh [hunk 3]
```diff
@@ -13,6 +13,9 @@ mkdir -p $cliHome
 echo "Copying files and plugins"
 cp -R $zipdir/* $cliHome
 
+echo "Creating corda-cli Script"
+echo "java -jar $cliHome/corda-cli.jar \"\$@\"" > $cliHome/corda-cli.sh
+
```
Comment (anton-subbotin) [inline tools/corda-cli/scripts/installScripts/install.sh:18]: See [other comment](https://github.com/corda/corda-runtime-os/pull/6264/files#r1666939413). 

To keep installer behaviour from 5.2, we need install scripts to generate run scripts as a part of the install process.

**Review by ronanbrowne — COMMENTED** [review]
Should the empty file `tools/corda-cli/scripts/runScripts/corda-cli.sh` be committed here?

**Comment by anton-subbotin** [comment]
> Should the empty file `tools/corda-cli/scripts/runScripts/corda-cli.sh` be committed here?

@ronanbrowne It's not empty, the change is only doing chmod 755 on that file to make it executable (which it hasn't been).
For some reason GH reports it as Empty file in the diff.

**Comment by ronanbrowne** [comment]
> > Should the empty file `tools/corda-cli/scripts/runScripts/corda-cli.sh` be committed here?
> 
> @ronanbrowne It's not empty, the change is only doing chmod 755 on that file to make it executable (which it hasn't been). For some reason GH reports it as Empty file in the diff.

Ah thanks Anton, yep that's clean now

**Review by ronanbrowne — APPROVED** [review]
LGTM

**Review by ghost — APPROVED** [review]
LGTM

**Pull request merged**